.github/workflows/tests.yml

```diff
@@ -9,19 +9,6 @@ on:
     - cron: '50 23 * * *'  # 11:50pm daily (to get ahead of all those midnight jobs)
 
 jobs:
-
-  linting:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check out repo
-        uses: actions/checkout@v4
-      - name: Install pre-commit
-        run: pip install pre-commit
-      - name: Run pre-commit
-        run: |
-          pre-commit install
-          pre-commit run --all
-
```

Comment:
So this linting job is now handled only by `pre-commit.ci`, which runs whenever PRs are opened or modified. But we are losing the periodic linting of the main branch because the `no-commit-to-branch` hook was failing when run against main. Is that right?

Reply:
Yes, I believe that is correct. A few options:
1. Don't worry about linting main periodically (current situation on this PR), pushes should have gone through a PR already because of branch protection rules.
2. Remove the no-commit-to-main check since that is the only one failing.
3. Instead of using the pre-commit.ci, use an explicit action/workflow: https://github.com/pre-commit/action (would still need to figure out how to skip (or remove) the commit-to-main check.

Reply:
Does no-commit-to-main even mean anything in the pre-commit.ci context? That commit has already been made and pushed if pre-commit.ci is running. If I understand correctly, the real utility of "no-commit-to-main" is to catch you if you accidentally forget to switch to a feature branch before you start working and you accidentally try to commit to `main` locally. I think I'm ok with skipping it for CI testing.